src/topology/sheaves/presheaf_of_functions.lean

```diff
@@ -34,6 +34,27 @@ namespace Top
 
 variables (X : Top.{v})
 
+/--
+The presheaf of dependently typed functions on `X`, with fibres given by a type family `f`.
+There is no requirement that the functions are continuous, here.
+-/
+def presheaf_to_Types (f : X → Type v) : X.presheaf (Type v) :=
+{ obj := λ U, Π x : (unop U), f x,
+  map := λ U V i g, λ (x : unop V), g (i.unop x) }
+
+/--
+The presheaf of functions on `X` with values in a type `T`.
+There is no requirement that the functions are continuous, here.
+-/
+def presheaf_to_Type (T : Type v) : X.presheaf (Type v) :=
+{ obj := λ U, (unop U) → T,
+  map := λ U V i g, λ (x : unop V), g (i.unop x) }
+
+@[simp] lemma presheaf_to_Type_map
```

Comment:
Is this the lemma that `@[simps]` would generate if you would write `g ∘ i.unop` in the previous definition? (incompatible with other comment on this file)

Reply:
Unfortunately not! Later it's convenient to have a simp lemma written as an equality of functions (rather than after applying to an argument).

I've added a comment explaining this.

Reply:
@semorrison: I think then `@[simps {fully_applied := ff}]` will do what you want. You might want to take a look at [docs#simp_cfg](https://leanprover-community.github.io/mathlib_docs/tactic/simps.html#simps_cfg).